x-pack/plugin/security/qa/basic-enable-security/build.gradle

```diff
@@ -25,11 +24,35 @@ testClusters {
     numberOfNodes = 2
     setting 'xpack.ml.enabled', 'false'
     setting 'xpack.license.self_generated.type', 'basic'
-    setting 'xpack.security.enabled', 'false'
```

Comment:
IIUC, the change here and other related ones adds a new test scenario for security-implicitly-disabled cluster. It's good to have broader coverage. But this wasn't covered before. Why is it necessary now when it's being deprecated? Is it just for the coverage?

Reply:
Ideally, I think, we would want to have two different configurations for each test: with and w/o security. For the sake of this specific change (which wasn't suppose to refactor entire elasticsearch test corpus) and as we discussed during our team meeting, I enable security wherever it is stright-forward and doesn't require an extra investigation on how these tests work and disable security in other cases.
Some tests are tickier, yaml tests would require additional refactering to be able to run with security on. As I said, I have a list of such tests and would love to either 1) us reviewing them in manageable portions and refactor them to run with security on 2) make specific area teams responsible for enabling security on their tests.
For the sake of this change I tried to keep it as simple as possible.

Reply:
As for this specific file - it is the test which intentionally runs with security implicitly disabled.

Reply:
Sorry maybe I wasn't being clear. My point is that we did not test for "security implicitly disabled" previously, so it seems unnecessary to test it now? But I don't have a strong opinion on it since the new test does have its value.

Reply:
Originally this QA test suite runs a few APIs on a cluster with basic license, with security implicitly disabled, and then it enables security and restarts the cluster to run similar APIs again and assert security is enabled.

I don't think adding a new transition in between, where security is explicitly disabled is valuable in this particular test suite.
Is the scenario this is supposed to cover, not covered elsewhere?
Alternatively, can we rely on randomization for when security is not enabled on basic license (either implicit or explicit)?

Not a fan of this approach, but I'll leave it in your hands.

Reply:
> Sorry maybe I wasn't being clear. My point is that we did not test for "security implicitly disabled" previously, so it seems unnecessary to test it now? But I don't have a strong opinion on it since the new test does have its value.

The purpose of this change is to add warning headers when security is implicitly disabled, so I added this new configuration to test the change ))